src/StakingNode.sol

```diff
@@ -257,23 +280,131 @@ contract StakingNode is IStakingNode, StakingNodeEvents, ReentrancyGuardUpgradea
         }
     }
 
+    /**
+     * @notice Queues multiple withdrawals for processing.
+     */
+    function queueWithdrawals(
+        uint256 sharesAmount
+    ) external onlyOperator returns (bytes32[] memory fullWithdrawalRoots) {
+
+        IDelegationManager delegationManager = IDelegationManager(address(stakingNodesManager.delegationManager()));
+
+        IDelegationManager.QueuedWithdrawalParams[] memory params = new IDelegationManager.QueuedWithdrawalParams[](1);
+        IStrategy[] memory strategies = new IStrategy[](1);
+        uint256[] memory shares = new uint256[](1);
+
+        strategies[0] = beaconChainETHStrategy;
+        shares[0] = sharesAmount;
+        params[0] = IDelegationManager.QueuedWithdrawalParams({
+            strategies: strategies,
+            shares: shares,
+            withdrawer: address(this)
+        });
+
+        fullWithdrawalRoots = delegationManager.queueWithdrawals(params);
+
+        queuedSharesAmount += sharesAmount;
+        emit QueuedWithdrawals(sharesAmount, fullWithdrawalRoots);
+    }
+
+    function completeQueuedWithdrawals(
+        IDelegationManager.Withdrawal[] memory withdrawals,
+        uint256[] memory middlewareTimesIndexes
+        ) external onlyOperator {
+
+        uint256 totalWithdrawalAmount = 0;
+
+        bool[] memory receiveAsTokens = new bool[](withdrawals.length);
+        IERC20[][] memory tokens = new IERC20[][](withdrawals.length);
+        for (uint256 i = 0; i < withdrawals.length; i++) {
+
+            // must be set to true even though what is received is native ETH
+            receiveAsTokens[i] = true;
+            // tokens array must match length of the withdrawals[i].strategies
+            // but does not need actual values in the case of the beaconChainETHStrategy
+            tokens[i] = new IERC20[](withdrawals[i].strategies.length);
+
+            for (uint256 j = 0; j < withdrawals[i].shares.length; j++) {
+                totalWithdrawalAmount += withdrawals[i].shares[j];
+            }
+        }
+
+        IDelegationManager delegationManager = IDelegationManager(address(stakingNodesManager.delegationManager()));
```

Comment:
Redundant casts. can simply call stakingNodesManager.delegationManager(). Also redundant state variable though, used only once.

Reply:
indeed lets simplify by removing the above line (412) and replacing line 422 with:

```solidity
stakingNodesManager.delegationManager().completeQueuedWithdrawals(
    withdrawals, 
    tokens, 
    middlewareTimesIndexes, 
    receiveAsTokens
);
```